components/script/dom/node.rs

```diff
@@ -284,6 +283,7 @@ impl Node {
 
         for node in child.traverse_preorder() {
             node.set_flag(IS_IN_DOC, false);
+            node.set_flag(HAS_DIRTY_DESCENDANTS, false);
```

Comment:
nit: You can probably `set_flag(IS_IN_DOC | HAS_DIRTY_DESCENDANTS, false)`.

Also, I think this deserves a comment.